From d2253b9768e3d6c6e4bf2a1890dc43d6a8fff3d5 Mon Sep 17 00:00:00 2001
From: Kristiyan Kostadinov <crisbeto@abv.bg>
Date: Fri, 1 Sep 2023 10:55:02 +0200
Subject: [PATCH] refactor(cdk/a11y): switch to input transforms

Switches inputs in cdk/a11y to use transforms instead of getters/setters.
---
 src/cdk/a11y/focus-trap/focus-trap.ts | 17 +++++------------
 tools/public_api_guard/cdk/a11y.md    | 10 ++++++----
 2 files changed, 11 insertions(+), 16 deletions(-)

diff --git a/src/cdk/a11y/focus-trap/focus-trap.ts b/src/cdk/a11y/focus-trap/focus-trap.ts
index 970c779ea2c0..9578898e9259 100644
--- a/src/cdk/a11y/focus-trap/focus-trap.ts
+++ b/src/cdk/a11y/focus-trap/focus-trap.ts
@@ -6,7 +6,6 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {BooleanInput, coerceBooleanProperty} from '@angular/cdk/coercion';
 import {_getFocusedElementPierceShadowDom} from '@angular/cdk/platform';
 import {DOCUMENT} from '@angular/common';
 import {
@@ -21,6 +20,7 @@ import {
   DoCheck,
   SimpleChanges,
   OnChanges,
+  booleanAttribute,
 } from '@angular/core';
 import {take} from 'rxjs/operators';
 import {InteractivityChecker} from '../interactivity-checker/interactivity-checker';
@@ -413,26 +413,19 @@ export class CdkTrapFocus implements OnDestroy, AfterContentInit, OnChanges, DoC
   private _previouslyFocusedElement: HTMLElement | null = null;
 
   /** Whether the focus trap is active. */
-  @Input('cdkTrapFocus')
+  @Input({alias: 'cdkTrapFocus', transform: booleanAttribute})
   get enabled(): boolean {
     return this.focusTrap.enabled;
   }
-  set enabled(value: BooleanInput) {
-    this.focusTrap.enabled = coerceBooleanProperty(value);
+  set enabled(value: boolean) {
+    this.focusTrap.enabled = value;
   }
 
   /**
    * Whether the directive should automatically move focus into the trapped region upon
    * initialization and return focus to the previous activeElement upon destruction.
    */
-  @Input('cdkTrapFocusAutoCapture')
-  get autoCapture(): boolean {
-    return this._autoCapture;
-  }
-  set autoCapture(value: BooleanInput) {
-    this._autoCapture = coerceBooleanProperty(value);
-  }
-  private _autoCapture: boolean;
+  @Input({alias: 'cdkTrapFocusAutoCapture', transform: booleanAttribute}) autoCapture: boolean;
 
   constructor(
     private _elementRef: ElementRef<HTMLElement>,
diff --git a/tools/public_api_guard/cdk/a11y.md b/tools/public_api_guard/cdk/a11y.md
index a2b136168ad7..42111b519039 100644
--- a/tools/public_api_guard/cdk/a11y.md
+++ b/tools/public_api_guard/cdk/a11y.md
@@ -6,7 +6,6 @@
 
 import { AfterContentInit } from '@angular/core';
 import { AfterViewInit } from '@angular/core';
-import { BooleanInput } from '@angular/cdk/coercion';
 import { ContentObserver } from '@angular/cdk/observers';
 import { DoCheck } from '@angular/core';
 import { ElementRef } from '@angular/core';
@@ -102,12 +101,15 @@ export class CdkMonitorFocus implements AfterViewInit, OnDestroy {
 export class CdkTrapFocus implements OnDestroy, AfterContentInit, OnChanges, DoCheck {
     constructor(_elementRef: ElementRef<HTMLElement>, _focusTrapFactory: FocusTrapFactory,
     _document: any);
-    get autoCapture(): boolean;
-    set autoCapture(value: BooleanInput);
+    autoCapture: boolean;
     get enabled(): boolean;
-    set enabled(value: BooleanInput);
+    set enabled(value: boolean);
     focusTrap: FocusTrap;
     // (undocumented)
+    static ngAcceptInputType_autoCapture: unknown;
+    // (undocumented)
+    static ngAcceptInputType_enabled: unknown;
+    // (undocumented)
     ngAfterContentInit(): void;
     // (undocumented)
     ngDoCheck(): void;
